src/realm/array.hpp

```diff
@@ -540,6 +540,16 @@ class Array : public ArrayParent {
     ///        this \c Array, sorted in ascending order
     /// \return the index of the value if found, or realm::not_found otherwise
     size_t find_gte(const int64_t target, size_t start, size_t end = size_t(-1)) const;
+
+    /// \brief Search the \c Array for a value greater or equal than \a target,
+    /// starting the search at the \a start index.
+    ///
+    /// \param target the smallest value to search for
+    /// \param start the offset at which to start searching in the array
+    /// \param end the offset at which to stop searching in the array
+    /// \return the index of the value if found, or realm::not_found otherwise
+    size_t find_gte_unsorted(const int64_t target, size_t start, size_t end = size_t(-1)) const;
```

Comment:
Just making it explicit: I think it would be a good idea to switch, even if there is no significant benefit on ARM. That way we'll easier benefit if/when we choose to take advantage of ARM vector instructions.